workflows/cloudify_system_workflows/snapshots/snapshot_restore.py

```diff
@@ -383,11 +384,8 @@ def _restore_credentials(self, postgres):
 
     def _restore_agents(self):
         ctx.logger.info('Restoring cloudify agent data')
-        Agents().restore(
-            self._tempdir,
-            self._tenant_name,
-            is_older_than_4_1_0=self._snapshot_version < V_4_1_0,
-        )
+        Agents().restore(self._tempdir,
+                         is_older_than_4_1_0=self._snapshot_version < V_4_1_0)
```

Comment:
Since you have an if statement for checking whether there are tenants and whether the version is earlier than 410 then perhaps have these as methods with a meaningful name, for example:
if `tenants_are_available` etc..